lib/runner.js

```diff
@@ -52,14 +52,20 @@ class Runner extends EventEmitter {
         return Promise.all(
             events.map(
                 ev =>
-                    new Promise(resolve =>
-                        this.once(ev, () => {
+                    new Promise((resolve, reject) => {
+                        this.once(ev, (succeeded) => {
+
                             const eventIndex = this.eventsToWait.indexOf(ev);
                             this.eventsToWait.splice(eventIndex, 1);
 
-                            return resolve();
+                            if (succeeded) {
```

Comment:
If we make the above mentioned change, this would check for an error during the execution of the events, like so:

```javascript
(err, result) => {
  if (err) {
    reject(err);
  } else {
    resolve(result);
  }
}
```

For now, no results are returned, but there may be some in the future, since this is supposed to be general-purpose, running different tasks. I know the runner is work in progress, but... baby steps :)